tests/test_import_export.py

```diff
@@ -0,0 +1,203 @@
+            with open(os.path.join(tempdir, "other.xml")) as f:
+                self.assertEqual(clean_xml(f.read()), clean_xml(input_xml))
+
+            handle_path = os.path.join(tempdir, "third.xml")
+            with open(handle_path, "w") as f:
+                bundle.save(file_handle=f)
+            with open(handle_path) as f:
+                self.assertEqual(clean_xml(f.read()), clean_xml(input_xml))
+        finally:
+            os.chdir(curdir)
+            rmtree(tempdir)
+
+    def test_export_and_keep_urls(self):
+        """
+        Test the changes to url_name after export_to_directory and import.
+        """
+        # Note url_name_orig in chapter.
+        input_xml = input_data.URL_NAME_ORIG_IN_CHAPTER2
+        bundle = XBundle(keep_urls=True, force_studio_format=True)
+        bundle.load(file_from_string(input_xml))
+
+        # str(bundle) doesn't change input xml, but export_to_directory will.
+        self.assertEqual(clean_xml(input_xml), clean_xml(str(bundle)))
+
+        old_current_dir = os.getcwd()
+        tempdir = mkdtemp()
+        try:
+            os.chdir(tempdir)
+            bundle.export_to_directory()
+
+            bundle2 = XBundle(keep_urls=True, force_studio_format=True)
+            bundle2.import_from_directory()
+
+            expected = expected_data.KEEP_URLS_FORCE_STUDIO_FORMAT
+
+            self.assertEqual(clean_xml(expected), clean_xml(str(bundle2)))
+        finally:
+            os.chdir(old_current_dir)
+            rmtree(tempdir)
+
+    def test_xml_header(self):
+        """
+        Test removal of xml attribute.
+        """
+        input_xml = input_data.EMPTY_XBUNDLE
+
+        bundle = XBundle()
+        bundle.load(file_from_string(input_xml))
+        self.assertEqual(clean_xml(input_xml), clean_xml(str(bundle)))
+
+    def test_import_skip_hidden(self):
+        """
+        Test skip_hidden flag.
+        """
+        bundle = XBundle(skip_hidden=True)
+        path = os.path.join('input_testdata', 'mitx.01')
+        bundle.import_from_directory(path)
+
+        expected = expected_data.SKIP_HIDDEN
+
+        self.assertEqual(clean_xml(str(bundle)), clean_xml(expected))
+
+    def test_import_large(self):
+        """
+        Test import of a course slightly larger than mitx.01.
+        """
+        bundle = XBundle()
+        path = os.path.join('input_testdata', 'content-devops-0001')
+        bundle.import_from_directory(path)
+
+        expected_path = os.path.join(
+            'input_testdata', 'content-devops-0001.out.xml')
+        with open(expected_path) as f:
+            self.assertEqual(clean_xml(f.read()), clean_xml(str(bundle)))
+
+        tempdir = mkdtemp()
+        try:
+            bundle.export_to_directory(tempdir, xml_only=True, newfmt=True)
+
+            for root, _, files in os.walk(os.path.join(tempdir, "0.001")):
+                for filename in files:
+                    abs_path = os.path.join(root, filename)
```

Comment:
No need to to any joining here; `filename` can be tested for `.endswith(".xml")`.